Feature/model monitoring

ranjan-stha opened the pull request:
Added scripts to calculate embeddings of excerpts, performance metrics at tag-level, per-project level, feature drift calculations per project.

bewakes committed:
Initial commit

bewakes committed:
Project setup: django, docker and celery

bewakes committed:
Setup workflow, test and coverage

bewakes committed:
Update README.md

bewakes committed:
WIP: Write core models. todo: update and makemigrate

bewakes committed:
Create functions for fetching deep data

bewakes committed:
Optimize data fetching

bewakes committed:
Update models

bewakes committed:
Upate models and enhance the fetch scripts

bewakes committed:
Add sentence transformers and app for deduplication

bewakes committed:
Update Models

bewakes committed:
Add scripts to create and save indices

bewakes committed:
Fix celery tasks issue

bewakes committed:
Merge migrations into one

bewakes committed:
Add deduplication api and corresponding configs

bewakes committed:
Add dedup-request processing tasks and test

bewakes committed:
Move all configs to environment

bewakes committed:
Fix tasks and schedules

bewakes committed:
Update docker compose, fix tasks

bewakes committed:
Update docker compose and fix some util functions

bewakes committed:
Fix re-indexing issue

bewakes committed:
Add tests for indexing task

bewakes committed:
Add gunicorn and prod docker compose configuration

bewakes committed:
Add lead_id to deep deuplication callback body

ranjan-stha committed:
scripts added for models monitoring;

ranjan-stha committed:
script updated; documentation added; cleanups;

ranjan-stha committed:
feature drift script added;

ranjan-stha committed:
script updated with date column and sample from dataset used to check for drift;

ranjan-stha committed:
added date column on computed metrics;

ranjan-stha committed:
date column added;

k9845 commented on monitoring_scripts/featuredrift.py:
```diff
@@ -0,0 +1,109 @@
+
+class FeatureDrift:
+    """
+    Calculates the per project feature drift of excerpts
```
```suggestion
    Calculates per project feature drift of excerpts
```

ranjan-stha replied:
done

k9845 commented on monitoring_scripts/featuredrift.py:
```diff
@@ -0,0 +1,109 @@
+    
+    def _project_id_based_mask(self,
+        df: pd.DataFrame,
+        project_id: float=None
```
Why do we have `project_id` of type float?

ranjan-stha replied:
fixed

k9845 commented on monitoring_scripts/featuredrift.py:
```diff
@@ -0,0 +1,109 @@
+                reference_df = pd.DataFrame(reference_embedding_lst).sample(n=n_samples, random_state=random_state)
+                current_df = pd.DataFrame(current_embedding_lst).sample(n=n_samples, random_state=random_state)
+
+                if len(reference_df) and len(current_df):
```
Shouldn't we check for length?

ranjan-stha replied:
fixed.

k9845 commented on monitoring_scripts/featuredrift.py:
```diff
@@ -0,0 +1,109 @@
+
+    feature_drift = FeatureDrift(reference_data_df, current_data_df)
+    df = feature_drift.compute_feature_drift(n_samples=10)
+    print(df)
```
```suggestion
```

k9845 commented on monitoring_scripts/featuredrift.py:
```diff
@@ -0,0 +1,109 @@
+                    temp_result["number_of_drifted_columns"] = data_drift_report["metrics"][0]["result"]["number_of_drifted_columns"]
+                    temp_result["share_of_drifted_columns"] = temp_result["drift_share"] = data_drift_report["metrics"][0]["result"]["share_of_drifted_columns"]
+                    temp_result["dataset_drift"] = temp_result["drift_share"] = data_drift_report["metrics"][0]["result"]["dataset_drift"]
+                    temp_result["generated_at"] = datetime.date.today()
+
+                    final_result.append(temp_result)
+        
+        return pd.DataFrame.from_records(final_result)
+    
+
+if __name__ == "__main__":
+    reference_data_path = "csvfiles/data_with_embeddings_22000.csv"
+    current_data_path = "csvfiles/sampled_data_with_embeddings_testset.csv"
```
This file path should be passed differently

ranjan-stha replied:
just used for testing.

k9845 commented on monitoring_scripts/generate_outputs.py:
```diff
@@ -0,0 +1,172 @@
+
+from postprocess_cpu_model_outputs import convert_current_dict_to_previous_one, get_predictions_all
+
+ENDPOINT_NAME = "main-model-cpu"
+AWS_REGION = "us-east-1"
+BATCH = 10
```
Let's pass this from `.env`

ranjan-stha replied:
reads from env

k9845 commented on monitoring_scripts/generate_outputs.py:
```diff
@@ -0,0 +1,172 @@
+        embeddings_required=True
+    )
+    df = embeddings.generate_outputs()
+    #print(df)
```
```suggestion
```

ranjan-stha committed:
updated script to get max samples of 500 from ref and current dataset;

ranjan-stha committed:
read values from env;

bewakes committed:
Update readme: add setup steps

ranjan-stha committed:
moved the model monitoring scripts;

bewakes committed:
Update deep callback payload

bewakes committed:
Enhance indexing of new project and leads; update migrations

bewakes committed:
Change dedup request response code to 202 accepted

bewakes committed:
Better modularize docker compose and update logging

bewakes committed:
Update github workflow: upload image to ECR

ranjan-stha committed:
Merge pull request #1 from the-deep-nlp/feature/deduplication

Feature/deduplication

bewakes committed:
Fix workflow yml file

bewakes committed:
Add staticfile config; rename docker volume names

bewakes committed:
Add script to run server and workers in a single ECS container

bewakes committed:
Update threshold for LSH indexing

ranjan-stha committed:
added the instructions and pkgs requirements;

ranjan-stha committed:
updated the instructions;

ranjan-stha committed:
requirements file updated;

ranjan-stha committed:
script updated; get the params while class obj is initialized;

bewakes committed:
Format nlp scripts

ranjan-stha committed:
added utils file; gets the model info;

ranjan-stha committed:
added the i/o descriptions;

bewakes committed:
Add embeddings field in entry

bewakes committed:
Update deep data fetch script: include sectors/pillars info

ranjan-stha committed:
updated the instructions;

Rup-Narayan-Rajbanshi committed:
Add a function to run nlp script

- Add fun to run generate_output.py and save resp data to model
- Add fun to run process_model.py
- Add github actions to run flake and black

Rup-Narayan-Rajbanshi committed:
Add a function to run nlp script

- Add fun to run generate_output.py and save resp data to model
- Add fun to run process_model.py

Rup-Narayan-Rajbanshi committed:
Add method to save model_performance into models

- Add models to save model_performance data
- Add function to save model_perf output to django model

ranjan-stha committed:
updated the example/script description;

ranjan-stha committed:
use literal_eval instead;

ranjan-stha committed:
preprocessing updates;

ranjan-stha committed:
supress warnings;

Rup-Narayan-Rajbanshi committed:
Schedule task to run model_monitoring

Rup-Narayan-Rajbanshi committed:
Modify admin site to show display list of model monitoring tables

Rup-Narayan-Rajbanshi committed:
Aggregated migration files for core

Rup-Narayan-Rajbanshi committed:
Set model monitoring task timing

- Add autocomplete fields

bewakes committed:
Add CSRF trusted origins setting
